planner/core/expression_rewriter.go

```diff
@@ -1132,6 +1132,11 @@ func (er *expressionRewriter) Leave(originInNode ast.Node) (retNode ast.Node, ok
 			} else {
 				castFunction.SetRepertoire(expression.UNICODE)
 			}
+
+			//See detailed comment in `expression/builtin_cast.go:WrapWithCastAsString()`; eg: "where CAST(bit as CAHR)"
+			if arg.GetType().Tp == mysql.TypeBit {
```

Comment:
Why we still need to add this if block after we added the same codes in the `WrapWithCastAsString`?

Reply:
Function ascii()、concat() will call through expression/builtin_cast.go::WrapWithCastAsString；
Function cast()、binary()  will call through planner/core/expression_rewriter.go::Leave;
The two kinds of functions be called not the same way,  so there are two positions needed to be updated.
And I don't know is there any other position need to update. So by copr-test, I only updated the 2 positions and passed the test;